lib/client/api_matrix.es6.js

```diff
@@ -87,24 +88,26 @@ foam.CLASS({
       code: function(releaseKeys, options) {
         options = options || {};
         let matrix = {};
-        let query = this.TRUE;
+
+        let query = this.IN(this.ReleaseWebInterfaceJunction.SOURCE_ID, releaseKeys);
         if (options.searchKey) {
-          query = this.AND(query, this.CONTAINS_IC(
-            this.WebInterface.INTERFACE_KEY, options.searchKey));
+          query = this.AND(
+              query,
+              this.CONTAINS_IC(this.ReleaseWebInterfaceJunction.TARGET_ID,
+                               options.seachKey));
         }
-        return Promise.all(releaseKeys.map(
-          (releaseId) => this.releaseDAO.find(releaseId).then((release) => {
-            // Reject if this release does not exit in database.
-            if (release === null) {
-              return Promise.reject(Error(`${releaseId} does not exist.`));
-            }
-            // TODO(markdittmer): ContextualizingDAO should take care of this.
-            return release.cls_.create(release, this.__subContext__)
-              .interfaces.dao.where(query).select(this.AnonymousSink.create({
-                sink: {put: iface => {
-                  let k0 = iface.interfaceName;
-                  let k1 = iface.apiName;
-                  let k2 = releaseId;
+
+        return this.releaseWebInterfaceJunctionDAO.where(query)
+            .select(this.AnonymousSink.create({
+              sink: {
+                put: junction => {
+                  var apiKeys = junction.targetId.split('#');
+                  foam.assert(apiKeys.length === 2,
+                              `Expected API key of the form:
+                                  Iface#propOrMethod`);
```

Comment:
The string "#" passes this check, so...you expect that the string have exactly one '#' character.